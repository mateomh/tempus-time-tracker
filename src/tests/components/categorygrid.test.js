import { createStore } from 'redux';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import CategoryGrid from '../../components/categorygrid';
import rootReducer from '../../reducers/root';

describe('CategoryGrid Component', () => {
  let mockStore;

  beforeEach(() => {
    const categories = [
      {
        name: 'Test 1',
        id: 123,
      },
      {
        name: 'Test 2',
        id: 456,
      },
    ];

    const mockState = {
      sessionUser: {},
      filter: {},
      categories,
      tasks: [],
      total: 0,
      error: '',
    };

    mockStore = createStore(rootReducer, mockState);
  });

  it(
    '1. Renders correctly',
    () => {
      render(
        <Provider store={mockStore}>
          <CategoryGrid />
        </Provider>,
      );
      let item = screen.getByText('Test 1');
      expect(item).toBeDefined();
      item = screen.getByText('Test 2');
      expect(item).toBeDefined();
    },
  );

  it(
    '2. Checks it renders correctly against the snapshot',
    () => {
      const tree = renderer
        .create(
          <Provider store={mockStore}>
            <CategoryGrid />
          </Provider>,
        ).toJSON();
      expect(tree).toMatchSnapshot();
    },
  );
});
