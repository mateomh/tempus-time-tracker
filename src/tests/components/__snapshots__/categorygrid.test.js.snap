// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryGrid Component 1. Checks it renders correctly against the snapshot 1`] = `
<div
  className="Container"
>
  <button
    className="Item"
    id={123}
    onClick={[Function]}
    type="button"
  >
    Test 1
  </button>
  <button
    className="Item"
    id={456}
    onClick={[Function]}
    type="button"
  >
    Test 2
  </button>
</div>
`;

exports[`CategoryGrid Component 2. Checks it renders correctly against the snapshot 1`] = `
<div
  className="Container"
>
  <button
    className="Item"
    id={123}
    onClick={[Function]}
    type="button"
  >
    Test 1
  </button>
  <button
    className="Item"
    id={456}
    onClick={[Function]}
    type="button"
  >
    Test 2
  </button>
</div>
`;
